antlr/DFDLGenerator.cpp

```diff
@@ -93,14 +92,14 @@ add_primitive_types(Element* schema)
 {
     add_primitive_type(schema, "float", "xs:float", "32");
     add_primitive_type(schema, "double", "xs:double", "64");
-    add_primitive_type(schema, "int8", "xs:integer", "8");
-    add_primitive_type(schema, "int16", "xs:integer", "16");
+    add_primitive_type(schema, "int8", "xs:byte", "8");
+    add_primitive_type(schema, "int16", "xs:short", "16");
     add_primitive_type(schema, "int32", "xs:integer", "32");
```

Comment:
Should this be `xs:int` https://www.w3.org/TR/xmlschema-2/#int ?

Reply:
Whoops. Yes, an xs:integer is a totally different thing.